lightkurve/lightcurve.py

```diff
@@ -541,33 +542,48 @@ def __repr__(self):
         elif self.mission.lower() == 'k2':
             return('KeplerLightCurveFile(EPIC: {})'.format(self.keplerid))
 
-    def correct(self, method='sff', **kwargs):
+    def correct(self, method='sff', warn=False, **kwargs):
         """Corrects a lightcurve for motion-dependent systematic errors.
 
         Parameters
         ----------
         method : str
             Method used to correct the lightcurve.
             Right now only 'sff' (Vanderburg's Self-Flat Fielding) is supported.
+        warn : bool
+            Turn off warnings for SFFCorrector
         kwargs : dict
             Dictionary of keyword arguments to be passed to the function
             defined by `method`.
 
+
         Returns
         -------
         new_lc : KeplerLightCurve object
             Corrected lightcurve
         """
+        ok = np.isfinite(self.flux)
         if method == 'sff':
-            self.corrector = SFFCorrector()
-            corrected_lc = self.corrector.correct(time=self.time, flux=self.flux,
-                                                  centroid_col=self.centroid_col,
-                                                  centroid_row=self.centroid_row,
-                                                  **kwargs)
+            if warn:
```

Comment:
One way to avoid the duplication of code may be something like this:
```
if not verbose:
    action = "default"
else:
   action = "ignore"
```
and then
```
warnings.simplefilter(action)
```

 with warnings.catch_warnings():